### PR TITLE
HTCONDOR-2689 Fix initial values for KeyboardIdle and LoadAvg in new …

### PR DESCRIPTION
**Pull request by johnkn**
…d-slots

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/param_info.in [hunk 1]
```diff
@@ -6109,6 +6109,96 @@ default : @end
...
+# Use a preemption policy that does cycle scavenging for Desktop computers with VACATE but not SUSPEND
+description=A preemption policy that runs jobs on workstations during non-work and non-workday hours
+default : @end
+    use FEATURE : POLICY_HOURS_TO_START($(1+))
+    # set CpuBusy latch and NonCondorLoadAvg in case admin wants to use them
+    NonCondorLoadAvg=(LoadAvg - CondorLoadAvg)
+    CpuBusy=($(NonCondorLoadAvg) >= $(HighLoad))
+    STARTD_LATCH_EXPRS = $(STARTD_LATCH_EXPRS) CpuBusy
+    # the CpuBusy LATCH_EXPR produces CpuBusyValue=0 or 1, CpuBusyTime is the time that CpuBusyValue changed
+    CpuBusyTimer=(CpuBusyValue * (time() - CpuBusyTime))
```
Comment (ColeBollig) [inline src/condor_utils/param_info.in:6185]: Are these needed for this desktop hours policy?

File: src/condor_utils/param_info.in [hunk 2]
```diff
@@ -6109,6 +6109,96 @@ default : @end
...
+       IsWeekend=ClockDay < $INT(WorkweekFirstDay) || ClockDay > $INT(WorkWeekLastDay)
+       IsWorkday=ClockDay >= $INT(WorkweekFirstDay) && ClockDay <= $INT(WorkWeekLastDay)
+       RunJobsHours=$(IsWeekend) || $(RunJobsHours)
+       NoJobsHours=$(IsWorkday) && $(NoJobsHours)
+    endif
+    RunJobsHours=($(RunJobsHours))
+    NoJobsHours=($(NoJobsHours))
+@end
+
+[$POLICY.DESKTOP_IDLE]
```
Comment (ColeBollig) [inline src/condor_utils/param_info.in:6137]: Do we need the regular desktop policy fragments for this policy?

File: src/condor_startd.V6/ResAttributes.cpp [hunk 3]
```diff
@@ -485,8 +485,11 @@ MachAttributes::compute_for_policy()
 	{ // formerly IS_TIMEOUT(how_much) && IS_SHARED(how_much)
 		m_load = sysapi_load_avg();
 
+
 		sysapi_idle_time( &m_idle, &m_console_idle );
 
+		//dprintf(D_ALWAYS | D_BACKTRACE, "MachAttributes::compute_for_policy idle=%d\n", m_idle);
+
```
Comment (ColeBollig) [inline src/condor_startd.V6/ResAttributes.cpp:492]: Do we want to keep this `dprintf`?

File: src/condor_startd.V6/ResAttributes.cpp [hunk 4]
```diff
@@ -2010,6 +2013,8 @@ CpuAttributes::publish_dynamic(ClassAd* cp) const
 		cp->Assign( ATTR_LOAD_AVG, rint((c_owner_load + c_condor_load) * 100) / 100.0 );
 		cp->Assign( ATTR_KEYBOARD_IDLE, c_idle );
 
+		// dprintf(D_IDLE, "KeyboardIdle=%lld LoadAvg=%.2f\n", c_idle, rint((c_owner_load + c_condor_load) * 100) / 100.0);
+
```
Comment (ColeBollig) [inline src/condor_startd.V6/ResAttributes.cpp:2017]: Do we want to keep this `dprintf`?

File: src/condor_utils/param_info.in (hunk 1 above)
Comment (johnkn) [reply src/condor_utils/param_info.in:6185]: No,  but they might be if someone wants to start with DESKTOP_HOURS and then add a load expression.

File: src/condor_startd.V6/ResAttributes.cpp (hunk 3 above)
Comment (johnkn) [reply src/condor_startd.V6/ResAttributes.cpp:492]: not sure,  prefer to leave it commented out for now.

File: src/condor_startd.V6/ResAttributes.cpp (hunk 4 above)
Comment (johnkn) [reply src/condor_startd.V6/ResAttributes.cpp:2017]: not sure prefer to leave it commented out for now.

File: src/condor_utils/param_info.in (hunk 2 above)
Comment (johnkn) [reply src/condor_utils/param_info.in:6137]: the pure hours policy should not load the policy fragments.

**Review by ColeBollig — APPROVED** [review]
LGTM

**Pull request merged**